roofit/roofit/inc/RooLagrangianMorphFunc.h

```diff
@@ -0,0 +1,367 @@
+  virtual std::list<Double_t> *
+  plotSamplingHint(RooAbsRealLValue & /*obs*/, Double_t /*xlo*/,
+                   Double_t /*xhi*/) const override;
+  virtual Bool_t isBinnedDistribution(const RooArgSet &obs) const override;
+  virtual Double_t evaluate() const override;
+  virtual TObject *clone(const char *newname) const override;
+  virtual Double_t getValV(const RooArgSet *set = 0) const override;
+
+  virtual Bool_t checkObservables(const RooArgSet *nset) const override;
+  virtual Bool_t forceAnalyticalInt(const RooAbsArg &arg) const override;
+  virtual Int_t
+  getAnalyticalIntegralWN(RooArgSet &allVars, RooArgSet &numVars,
+                          const RooArgSet *normSet,
+                          const char *rangeName = 0) const override;
+  virtual Double_t
+  analyticalIntegralWN(Int_t code, const RooArgSet *normSet,
+                       const char *rangeName = 0) const override;
+  virtual void printMetaArgs(std::ostream &os) const override;
+  virtual RooAbsArg::CacheMode canNodeBeCached() const override;
+  virtual void setCacheAndTrackHints(RooArgSet &) override;
+
+  void insert(RooWorkspace *ws);
+
+  void setParameters(const char *foldername);
+  void setParameters(TH1 *paramhist);
+  void setParameter(const char *name, double value);
+  void setFlag(const char *name, double value);
+  void setParameters(const ParamSet &params);
+  void setParameters(const RooArgList *list);
+  double getParameterValue(const char *name) const;
+  RooRealVar *getParameter(const char *name) const;
+  RooRealVar *getFlag(const char *name) const;
+  bool hasParameter(const char *paramname) const;
+  bool isParameterUsed(const char *paramname) const;
+  bool isParameterConstant(const char *paramname) const;
+  void setParameterConstant(const char *paramname, bool constant) const;
+  void setParameter(const char *name, double value, double min, double max);
+  void setParameter(const char *name, double value, double min, double max,
+                    double error);
+  void randomizeParameters(double z);
+  const RooArgSet *getParameterSet() const;
+  using RooAbsArg::getParameters;
+  ParamSet getParameters(const char *foldername) const;
+  ParamSet getParameters() const;
+
+  RooLagrangianMorphFunc *getLinear() const;
+
+  int nParameters() const;
+  int nPolynomials() const;
+
+  bool isCouplingUsed(const char *couplname);
+  const RooArgList *getCouplingSet() const;
+  ParamSet getCouplings() const;
+
+  // virtual Bool_t IsAllowNegativeYields() const { return
+  // this->_config.IsAllowNegativeYields(); }
+  TMatrixD getMatrix() const;
+  TMatrixD getInvertedMatrix() const;
+  double getCondition() const;
+
+  RooRealVar *getObservable() const;
+  RooRealVar *getBinWidth() const;
+
+  void printEvaluation() const;
+  void printCouplings() const;
+  void printFlags() const;
+  void printPhysics() const;
+
+  RooProduct *getSumElement(const char *name) const;
+
+  const std::vector<std::string> getSamples() const;
```

Comment:
```suggestion
  std::vector<std::string> getSamples() const;
```
A const qualifier is not meaningful if you return a value.